src/main/java/org/commonwl/view/cwl/CWLTool.java

```diff
@@ -126,14 +130,20 @@ private String runCwltoolOnWorkflow(String argument, String workflowUrl)
       int exitCode = process.waitFor();
       if (exitCode == 0) {
         inputGobbler.join();
+        process.destroyForcibly();
         return inputGobbler.getContent();
       } else {
         errorGobbler.join();
+        process.destroyForcibly();
         throw new CWLValidationException(errorGobbler.getContent());
       }
     } catch (IOException | InterruptedException e) {
       logger.error("Error running cwltool process", e);
       throw new CWLValidationException("Error running cwltool process");
+    } finally {
+      if (process != null) {
+        process.destroyForcibly();
+      }
```

Comment:
Ha, you have a finally with the `process` here. So maybe apply this to the other one too?

Reply:
Yes I agree. For now it is probably not necessary, but putting the logic directly inside a `finally` clause avoids potential regressions in the future in case the code changes.